bootstrap_salt/fab_tasks.py

```diff
@@ -100,211 +105,121 @@ def application(application_name):
     env.application = str(application_name).lower()
 
 
-@task
-def setup(salt_version='v2014.7.5'):
+def get_kms_key_id():
     """
-    Setup the salt master and minions
-
-    Call install_master and install_minions to setup salt
+    Returns an id for a KMS master key with an alias of:
+    application-environment. If no key exists it will create a
+    new one.
     """
-    install_master(salt_version)
-    install_minions(salt_version)
+    alias = "{0}-{1}".format(env.application, env.environment)
+    kms = get_connection(KMS)
+    key_id = kms.get_key_id(alias)
+    if not key_id:
+        key_id = kms.create_key(alias)
+    return key_id
```

Comment:
Does this mean multiple stacks for the same env share a KMS key and salt s3 bucket? (i.e. the case when we are migrating an environment to a newly built stack)